mariamihaelafainarea/assignment3/Assignment3

```diff
@@ -120,26 +120,22 @@ public class Assignment3 {
             current.endOfWord = true;
             
         }
-        public boolean isGeneralHelper(String action,TrieNode current,String word,int index) {
-            if(action.equals("word")) {
+        private boolean isGeneralHelper(boolean action,TrieNode current,String word,int index) {
```

Comment:
Name "action" in a more clear way. Right now I would ask you to document what "action == true" means but it's even better to just name it "isWord" or "wordCheck" or something like that. So that it is clear that if it is true you are checking for a word not a prefix.

Reply:
Okey. i  have named action -> isWord

Reply:
i want to say that the comment is over an old version of the problem. Before me modifying the action parameter the generalHelper starts with :
        private boolean isGeneralHelper(boolean action,TrieNode current,String word,int index) {
            if(action) {
                if (index == word.length()) {
                    return current.endOfWord;
                }
            } else {
                if (index == word.length()) {
                    return current != null;
                }
            }